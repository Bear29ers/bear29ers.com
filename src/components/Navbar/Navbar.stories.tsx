import Navbar from '@/components/Navbar/Navbar';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Navbar> = {
  component: Navbar,
};

export default meta;

type Story = StoryObj<typeof Navbar>;

export const Standard: Story = {
  render: () => <Navbar />,
};
