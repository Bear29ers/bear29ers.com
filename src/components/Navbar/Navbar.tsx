'use client';

import type { FC } from 'react';

import { motion } from 'framer-motion';

import { NAV_CONTENTS, SOCIAL_LINK } from '@/constants';

import FacebookIcon from '../Icons/FacebookIcon/FacebookIcon';
import GithubIcon from '../Icons/GithubIcon/GithubIcon';
import InstagramIcon from '../Icons/InstagramIcon/InstagramIcon';
import Logo from '../Logo/Logo';
import NavLink from '../NavLink/NavLink';

const Navbar: FC = () => {
  return (
    <header className="flex-between w-full px-32 py-8 font-medium" role="banner">
      <nav className="space-x-4">
        {NAV_CONTENTS.map((content) => (
          <NavLink href={content.href} title={content.title} key={content.id} />
        ))}
      </nav>
      <nav className="flex-center flex-wrap space-x-3">
        <motion.a
          href={SOCIAL_LINK.github}
          target="_blank"
          className="w-6"
          whileHover={{ y: -2 }}
          whileTap={{ scale: 0.9 }}>
          <GithubIcon />
        </motion.a>
        <motion.a
          href={SOCIAL_LINK.facebook}
          target="_blank"
          className="w-6"
          whileHover={{ y: -2 }}
          whileTap={{ scale: 0.9 }}>
          <FacebookIcon />
        </motion.a>
        <motion.a
          href={SOCIAL_LINK.instagram}
          target="_blank"
          className="w-6"
          whileHover={{ y: -2 }}
          whileTap={{ scale: 0.9 }}>
          <InstagramIcon />
        </motion.a>
      </nav>
      <div className="absolute left-1/2 top-2 translate-x-1/2">
        <Logo />
      </div>
    </header>
  );
};

export default Navbar;
