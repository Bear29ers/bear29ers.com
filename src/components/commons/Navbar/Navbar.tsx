'use client';

import type { FC } from 'react';

import { motion } from 'framer-motion';

import { NAV_CONTENTS } from '../../../constants/nav';
import { SOCIAL_MEDIA_LIST } from '../../../constants/socialMedia';
import Logo from '../Logo/Logo';
import NavLink from '../NavLink/NavLink';

const Navbar: FC = () => {
  return (
    <header className="flex-between w-full py-8 font-medium" role="banner">
      <nav className="space-x-4">
        {NAV_CONTENTS.map((content) => (
          <NavLink href={content.href} title={content.title} key={content.id} />
        ))}
      </nav>
      <nav className="flex-center flex-wrap space-x-3">
        {SOCIAL_MEDIA_LIST.map((socialMedia) => (
          <motion.a
            href={socialMedia.href}
            target="_blank"
            className="w-6"
            whileHover={{ y: -2 }}
            whileTap={{ scale: 0.9 }}
            role="link"
            aria-describedby={socialMedia.describedby}
            key={socialMedia.id}>
            <socialMedia.component />
            <div id={socialMedia.describedby} className="hidden">
              {socialMedia.text}
            </div>
          </motion.a>
        ))}
      </nav>
      <div className="absolute left-1/2 top-2 translate-x-1/2">
        <Logo />
      </div>
    </header>
  );
};

export default Navbar;
