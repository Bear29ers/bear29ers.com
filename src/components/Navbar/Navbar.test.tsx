import { render } from '@testing-library/react';

import Navbar from '@/components/Navbar/Navbar';

describe('src/components/Navbar/Navbar.test.tsx', () => {
  it('renders correctly', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
