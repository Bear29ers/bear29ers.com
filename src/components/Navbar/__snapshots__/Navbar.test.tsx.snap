// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Navbar/Navbar.test.tsx renders correctly 1`] = `
<div>
  <header
    class="flex w-full items-center justify-between px-32 py-8 font-medium"
  >
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about"
      >
        About
      </a>
      <a
        href="/projects"
      >
        Projects
      </a>
      <a
        href="/articles"
      >
        Articles
      </a>
    </nav>
    <h2>
      Logo
    </h2>
    <nav>
      <a
        href="/"
        target="_brank"
      >
        T
      </a>
      <a
        href="/"
        target="_brank"
      >
        T
      </a>
      <a
        href="/"
        target="_brank"
      >
        T
      </a>
      <a
        href="/"
        target="_brank"
      >
        T
      </a>
      <a
        href="/"
        target="_brank"
      >
        T
      </a>
      <a
        href="/"
        target="_brank"
      >
        T
      </a>
    </nav>
  </header>
</div>
`;
